platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/command/AbstractProfileCommand.java

```diff
@@ -0,0 +1,186 @@
+    Boolean showDevProfiles = false;
+
+    Path developmentProfilePath = Paths.get(System.getProperty("ddf.home"), "etc", "admin", "profiles");
+
+    String DEV_PROFILE_INSTALL_FEATURES = "install-features";
+    String DEV_PROFILE_UNINSTALL_FEATURES = "uninstall-features";
+    String DEV_PROFILE_START_APPS = "start-apps";
+    String DEV_PROFILE_STOP_BUNDLES = "stop-bundles";
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    /**
+     * Check if an installation profile exists
+     * @param installProfiles the list of installation profiles
+     * @param profileName the name of the profile to check for
+     * @return {@link Boolean} profile exists?
+     */
+    Boolean profileExists(List<Feature> installProfiles, String profileName) {
+        return installProfiles.stream().anyMatch(feature -> feature.getName().contains(profileName));
+    }
+
+    /**
+     * Check if a dev profile exists (Experimental)
+     * @param profileName name of profile to check for
+     * @return {@link Boolean} profile exists?
+     */
+    Boolean devProfileExists(String profileName) {
+        return Paths.get(developmentProfilePath.toString(), profileName + ".json").toFile().exists();
+    }
+
+    /**
+     * Get the {@link Feature} that makes up a profile
+     * @param applicationService {@link ApplicationService}
+     * @param profileName Name of profile to get
+     * @return {@link Feature}
+     */
+    Feature getProfile(ApplicationService applicationService, String profileName) {
+        return applicationService.getInstallationProfiles().stream()
+                .filter(i -> i.getName().contains(profileName))
+                .findFirst()
+                .get();
+    }
+
+    /**
+     * Gets the content of a dev profile
+     * @param profileName Name of profile to get
+     * @return {@link Map}
+     */
+    Map<String, List<String>> getDevProfile(String profileName) {
+        FileInputStream fileStream;
+        File devProfileFile = Paths.get(developmentProfilePath.toString(), profileName + ".json").toFile();
+        if (!isValidJsonFile(devProfileFile)) {
+            return null;
+        }
+
+        try {
+            fileStream = new FileInputStream(devProfileFile);
+        } catch (FileNotFoundException e) {
+            logger.debug("File not found.");
+            return null;
+        }
+
+        try {
+            return (Map<String, List<String>>) fromJson(IOUtils.toString(fileStream));
+        } catch (IOException e) {
+            logger.debug("An IOException occurred: ", e);
+            return null;
+        } finally {
+            try {
+                fileStream.close();
+            } catch (IOException e) {
+                logger.debug("An IOException occurred: ", e);
+                return null;
+            }
+        }
+    }
+
+    private boolean isValidJsonFile(File configFile) {
```

Comment:
I concur. You're only invoking this locally after you've read in a file specifically ending in the `.json` extension. If it exists, then it has that extension. That would leave this method to check for file existence which doesn't give it much value.